scene/3d/physics_body_3d.cpp

```diff
@@ -2510,7 +2591,10 @@ PhysicalBone3D::PhysicalBone3D() :
 		bounce(0),
 		mass(1),
 		friction(1),
-		gravity_scale(1) {
+		gravity_scale(1),
+		linear_damp(-1),
+		angular_damp(-1),
```

Comment:
This should be 0.

Reply:
The default value of -1 is the same as for `RigidBody3D`. This value means it uses the world/area value, and a value >=0 means it's overriden for the rigid body. It's working this way in Godot Physics but it seems it's not implemented the same way in Bullet. It's probably something to be fixed as a separate PR.

Reply:
Since damping should be always positive, it's default value can't be negative, even if it's clamped internally.

Reply:
It's true that inside of the engine implementation the damping needs to be >=0. However, in the editor, there's a difference between setting -1 (use area/world default value) and 0 (use no damping).

That's why `RigidBody3D` already uses -1 as a default value for damping:
https://github.com/godotengine/godot/blob/e0eb9a8ae474ad8ed2a155b176e0c1b14cf66773/scene/3d/physics_body_3d.cpp#L916-L917

And here's the Godot Physics code that handles this logic:
(the comment doesn't seem accurate though)
https://github.com/godotengine/godot/blob/e0eb9a8ae474ad8ed2a155b176e0c1b14cf66773/servers/physics_3d/body_3d_sw.cpp#L487-L506

Reply:
We are exposing an inner technical detail, of a specific physics server.

The API `set_damping` is just a simple function to set the body damping, and as so it should be treated.

Building a functionality around a _specific server_ feature is not a good idea, and is not the usual way we take.

I know it was already so, but I think that is better set the default damping to 0, so we can fulfill what the docs says, and later add a new API to change how the damping is applied.

Reply:
Also, I think that the damping should not be combined in that way. Because doing so the space override specified by the area are not respected:
https://github.com/godotengine/godot/blob/e0eb9a8ae474ad8ed2a155b176e0c1b14cf66773/servers/physics_3d/body_3d_sw.cpp#L468-L480

The area is overriding the body value and not vice versa.

Reply:
For example:
I set body damp to 0.5 and I want that the area overrides it with 1 when the body is inside it.

With the above code this would not be possible because the area override is overridden by the body (since 0.5 so greater than 0).

Reply:
Ok, I get it! That makes sense if we decide areas can override body damping values and not the other way around. It looks like a good idea to me.

In this case, in order to keep things consistent, we need to change more things:
- Not allow damping to be negative in the editor
- Make damping non-negative and default to 0 for `RigidBody` as well
- Update `RigidBody` documentation to reflect this change
- Modify Godot Physics implementation to make area values override body values

I would do the `RigidBody` and Godot Physics changes in a separate PR if they seem ok to you.

Reply:
yep